sdk/iOS/src/MSUser.h

```diff
@@ -18,7 +18,7 @@
 ///@{
 
 /// Initializes an *MSUser* instance with the given user id.
--(id)initWithUserId:(NSString *)userId;
+-(nonnull instancetype)initWithUserId:(nonnull NSString *)userId;
```

Comment:
We have a unit test: -(void) testInitWithUserIdAllowsNilUserId, that says you can use nil here.  To me, since the auth token is the only important data it makes sense to allow no user Id (Its been awhile, but I don't remember caching my userIds in the past when building auth out)